src/main/java/org/fenixedu/academic/domain/degreeStructure/Context.java

```diff
@@ -134,6 +128,7 @@ public Context(final CourseGroup courseGroup, final DegreeModule degreeModule, f
         super.setParentCourseGroup(courseGroup);
         super.setChildDegreeModule(degreeModule);
         super.setCurricularPeriod(curricularPeriod);
+        super.setTerm(term == null || term.intValue() < 1 ? null : term);
```

Comment:
shouldn't this return an error message if term < 1?

Reply:
No. The term is optional. If the context represents a year or a semester, then there is no term.